src/riscv/lib/src/pvm/tezos.rs

```diff
@@ -359,6 +359,10 @@ pub(super) fn handle_tezos<MC, M>(
     MC: MemoryConfig,
     M: ManagerReadWrite,
 {
+    // TODO: RV-777: remove below and instead have each system call return a `ProgramCounterUpdate`
+    let pc = machine.hart.pc.read().saturating_add(4);
```

Comment:
There is a bug in the original code. It should be wrapping addition. Would you mind changing that?